### ISSUE TITLE
VideoReader get_frame_timestamp incorrect

### ISSUE THREAD

**Issue by sbebo** [dmlc/decord#52]
It looks like `get_frame_timestamp` is returning timestamps which are correct but non always consistent with what is returned by `__getitem__` (obv this is an issue only in videos where frames are decoded in a different order from the presentation order).

More precisely, it looks like `vr.get_frame_timestamp(i)` is returning the pts of the `i`-th frame ordering by **_dts_**, while `vr[i]` would return the `i`-th frame ordering by **_pts_** time.

Example:

```python
> vr.get_frame_timestamp(range(5))                                                             
array([[0.        , 0.03333334],
       [0.03333334, 0.06666667],
       [0.01333333, 0.04666667],
       [0.1       , 0.13333334],
       [0.06666667, 0.1       ]], dtype=float32)
```
while `vr[:5]` will correctly return the first 5 frames ordered by presentation time.

This is the output of an `ffprobe -show_packets -select_streams v:0 `on the file:
```
[PACKET]
codec_type=video
stream_index=0
pts=0
pts_time=0.000000
dts=-672
dts_time=-0.035000
duration=640
duration_time=0.033333
convergence_duration=N/A
convergence_duration_time=N/A
size=50364
pos=36
flags=K_
[/PACKET]
[PACKET]
codec_type=video
stream_index=0
pts=640
pts_time=0.033333
dts=-671
dts_time=-0.034948
duration=640
duration_time=0.033333
convergence_duration=N/A
convergence_duration_time=N/A
size=24018
pos=50400
flags=__
[/PACKET]
[PACKET]
codec_type=video
stream_index=0
pts=256
pts_time=0.013333
dts=224
dts_time=0.011667
duration=640
duration_time=0.033333
convergence_duration=N/A
convergence_duration_time=N/A
size=9179
pos=74418
flags=__
[/PACKET]
[PACKET]
codec_type=video
stream_index=0
pts=1920
pts_time=0.100000
dts=608
dts_time=0.031667
duration=640
duration_time=0.033333
convergence_duration=N/A
convergence_duration_time=N/A
size=18722
pos=83597
flags=__
[/PACKET]
[PACKET]
codec_type=video
stream_index=0
pts=1280
pts_time=0.066667
dts=1248
dts_time=0.065000
duration=640
duration_time=0.033333
convergence_duration=N/A
convergence_duration_time=N/A
size=8483
pos=102319
flags=__
[/PACKET]

```

**Comment by zhreshold** [comment]
I think it's the correct pts but the problem might be that I forgot to sort the pts, the ffmpeg is handling the reordering of the pts correctly.

**Comment by sbebo** [comment]
Yeah pts are correct, just in the "wrong" order (ordered by dts). Just sorting before returning should fix it.

**Comment by zhreshold** [comment]
closed by #53
